### ISSUE TITLE
No type hints for `Tensor.T`

### ISSUE THREAD

**Issue by henribru** [pytorch/pytorch#28457]
## 🐛 Bug

There are no typehints for `Tensor.T`, so MyPy can't tell it exists.
## To Reproduce

1. Define `test.py` as:
```py
import torch

torch.tensor([[1, 2], [3, 4]]).T
```
1. Run `mypy test.py`:
```
$ mypy test.py
test.py:3: error: "Tensor" has no attribute "T"
Found 1 error in 1 file (checked 1 source file)
```

## Expected behavior
The file should not give a type error.

## Environment

```
$ python collect_env.py 
Collecting environment information...
PyTorch version: 1.2.0
Is debug build: No
CUDA used to build PyTorch: 10.0.130

OS: Ubuntu 18.04.3 LTS
GCC version: (Ubuntu 7.4.0-1ubuntu1~18.04.1) 7.4.0
CMake version: version 3.10.2

Python version: 3.6
Is CUDA available: Yes
CUDA runtime version: 10.1.243
GPU models and configuration: 
GPU 0: Tesla V100-PCIE-32GB
GPU 1: Tesla V100-PCIE-32GB

Nvidia driver version: 430.26
cuDNN version: /usr/lib/x86_64-linux-gnu/libcudnn.so.7.4.2

Versions of relevant libraries:
[pip3] numpy==1.17.2
[pip3] torch==1.2.0
[pip3] torchvision==0.4.0
[conda] Could not collect
```
## Additional context

cc @ezyang

**Comment by fmassa** [comment]
We would be happy to accept a PR fixing this

**Comment by henribru** [comment]
I had a look but found the mix of manual and generated type hints a bit confusing. Is this a simple matter of adding it in the stub file or does the generation need to be changed?

**Comment by fmassa** [comment]
@t-vi might know better, I'm not sure of the details for the state of type hints in PyTorch

**Comment by t-vi** [comment]
I hope to not do know, but as a rule of thumb, only the things generated from `native_functions.yaml` are autogenerated. `.T` is defined manually `python_variable.cpp` below, so the hints would be manual, too.
https://github.com/pytorch/pytorch/blob/ee920b92c4d52c3481ce4fa1594893fde8ccdd93/torch/csrc/autograd/python_variable.cpp#L491

**Comment by henribru** [comment]
^ I made a PR for this.

**Comment by t-vi** [comment]
@henribru Francisco already beat me to approving and is landing your PRs, but let me say: thank you!

As a nit: Note that the top message of the PR will be the commit message, so cross-referencing unrelated PRs will be recorded for eternity. (Not that I always get the commit messages as good as I want them.)

**Comment by avivrosenberg** [comment]
This issue was closed a while ago.
However, I'm getting this error (`"Tensor" has no attribute "T"`) with `pytorch==1.6.0` and `mypy==0.782`.

Any ideas?

**Comment by ez4lionky** [comment]
I also met the error with pytorch==1.1.0, but it's normal in 1.4.0, any idea?

**Comment by t-vi** [comment]
Yeah, in 1.1.0, type hints weren't as complete as they are now.

**Comment by ez4lionky** [comment]
> Yeah, in 1.1.0, type hints weren't as complete as they are now.
I also think it should be a version issue. :)

> This issue was closed a while ago.
> However, I'm getting this error (`"Tensor" has no attribute "T"`) with `pytorch==1.6.0` and `mypy==0.782`.
> 
> Any ideas?

**Comment by avivrosenberg** [comment]
Perhaps it worked in `1.4.0`, I haven't tested. But as I mentioned, I see this now constantly with with `pytorch==1.6.0` and `mypy==0.782`.

**Comment by t-vi** [comment]
Yeah well, apparently some things were lost when the .pyi was reorganized. It was then reported as #40658 and fixed in July, but likely after 1.6 was cut.
